### ISSUE TITLE
Unable to get started using ARM Overlay

### ISSUE THREAD

**Issue by mutemule** [rancher/k3os#451]
**Version (k3OS / kernel)**
k3os: v0.10.0
kernel: 4.9.140

**Architecture**
aarch64

This is using an nVidia Jetson Nano developer board.

**Describe the bug**
Following the instructions to install k3os using the ARM Overlay approach, the resulting system boots to a login prompt, but doesn't appear to be configured as per the provided `config.yaml`: I cannot log in using the provided password, and the hostname remains `k3os-nnnnn`.

**To Reproduce**
1. Image a microSD card using nVidia's Jetson Nano image. Go through the initial setup, resize the filesystem, add a user, etc., until it reboots to the login prompt.
2. Swap to a text console.
3. Log in as the user you specified during setup.
4. Become `root`.
5. Create a new partition, #15 (yes, the image does have partitions 1-14 created), a 4B in size, as type Linux Filesystem.
6. Create an ext4 filesystem labelled `K3OS_STATE`: `mkfs -L K3OS_STATE /dev/mmcblk0p15`
7. Download and extract the root filesystem as instructed:
```curl -sfL https://github.com/rancher/k3os/releases/download/v0.10.0/k3os-rootfs-arm.tar.gz | tar zxvf - --strip-components=1 -C /```
8. Create a minimal configuration file at `/k3os/system/config.yaml`:
```
hostname: jetson1

k3os:
  k3s_args:
  - server
  password: dr0wssap
  token: jetsonnanodemo

run_cmd:
- rc-service ntpd start
- sleep 5

ssh_authorized_keys:
- github:mutemule
```
9. Run `sync`
10. Run `shutdown -r now`
11. Wait for k3os to boot.

**Expected behavior**
1. k3os should boot 
2. Appropriate first-run initialization should happen
3. I should be able to login as `rancher` with the password `drw0wssap`
4. The hostname should be `jetson1`

**Actual behavior**
1. k3os boots to a login prompt and gets an IP, but:
  * the password for `rancher` is changed twice, not the once that was expected
  * `sysctl` init fails, as some keys don't exist (`net.ipv4.tcp_syncookies` and `kernel.unprivileged_bpf_disabled`) -- note that this happens twice
  * `iscsid` fails to start (it is unexpected that it even tried to start, but not surprising that it failed to do so)
2. There is a section of log that indicates k3s v1.17.4+k3s1 is installed, and a handful of symlinks are created. This seems to be about right.
3. When I try to log in as `rancher` on `tty1`, the word `rancher` is echo'd back, then I am prompted for the password, but the system doesn't appear to accept any typing. If I swap to `tty2`, all logins as `rancher` and `root` fail. Attempts to log in to the machine via SSH are rejected without ever being prompted for a password:
```
rancher@172.19.20.118: Permission denied (publickey,keyboard-interactive).
```
4. The hostname is `k3os-13733`

At this point, because I have no credentials, I can't access the running system. I can pull the microSD card and mount it elsewhere to look at the resulting filesystem, but that hasn't revealed anything obvious.

**Additional context**
I initially tried this without creating the `K3OS_STATE` partition, with the same symptoms, as there's nothing in the ARM Overlay instructions about creating that partition.

The kernel is, sadly, stuck at 4.9.140, as that's the latest version supported by the [Linux4Tegra ](https://developer.nvidia.com/embedded/linux-tegra)project. While woefully out of date, the kernel should be sufficient to run k3os.

IPv4 negotiation via DHCP is working just fine: the system is online, I can ping it, but a port scan shows that only TCP/22 is listening

This feels like it might be related to #448.

**Comment by PhoenixMage** [comment]
I'd triple check your filename, I had almost the exact smae behavious as I had /k3os/system/config.yml rather then /k3os/system/config.yaml

**Comment by qtvhao** [comment]
Yes, I have similar problem: *When I try to log in as rancher on tty1, the word rancher is echo'd back, then I am prompted for the password, but the system doesn't appear to accept any typing. If I swap to tty2, all logins as rancher and root fail. Attempts to log in to the machine via SSH are rejected without ever being prompted for a password.*

But then k3s-killall.sh solve problem.
